File: app/src/main/java/cn/invonate/ygoa3/Util/TimeUtil.java
package cn.invonate.ygoa3.Util;

import java.text.SimpleDateFormat;
import java.util.Date;

public class TimeUtil {
    public static String timeFormatWithDoday(long time) {
        StringBuffer pattern = new StringBuffer();
        Date today = new Date();
        Date receivedDate = new Date(time);
        if (today.getYear() != receivedDate.getYear()) {
            pattern.append("yyyy-");
        }
        if (today.getYear() == receivedDate.getYear() && today.getMonth() == receivedDate.getMonth() && today.getDate() - 1 == receivedDate.getDate()) {
            pattern.append("昨天");
        }

        if (today.getYear() == receivedDate.getYear() && today.getDate() - 1 != receivedDate.getDate() && today.getDate() != receivedDate.getDate()) {
            pattern.append("MM-dd ");
        }
        pattern.append("HH:mm");
        return new SimpleDateFormat(pattern.toString()).format(receivedDate);

    }

    public static String timeFormatNoYear(long time) {
        StringBuffer pattern = new StringBuffer();
        pattern.append("MM:mm EE HH:mm");
        return new SimpleDateFormat(pattern.toString()).format(new Date(time));
    }

    public static String timeFormatJustMMHH(long time) {
        StringBuffer pattern = new StringBuffer();
        pattern.append("HH:mm");
        return new SimpleDateFormat(pattern.toString()).format(new Date(time));
    }
}

File: app/src/main/java/cn/invonate/ygoa3/Util/TimeUtil.kt
package cn.invonate.ygoa3.Util

import java.text.SimpleDateFormat
import java.util.*

object TimeUtil {
    fun timeFormatWithDoday(time: Long): String {
        val pattern = StringBuffer()
        val today = Date()
        val receivedDate = Date(time)
        if (today.year != receivedDate.year) {
            pattern.append("yyyy-")
        }
        if (today.year == receivedDate.year && today.month == receivedDate.month && today.date - 1 == receivedDate.date) {
            pattern.append("昨天")
        }

        if (today.year == receivedDate.year && today.date - 1 != receivedDate.date && today.date != receivedDate.date) {
            pattern.append("MM-dd ")
        }
        pattern.append("HH:mm")
        return SimpleDateFormat(pattern.toString()).format(receivedDate)

    }

    fun timeFormatNoYear(time: Long): String {
        val pattern = StringBuffer()
        pattern.append("MM-dd EE HH:mm")
        return SimpleDateFormat(pattern.toString()).format(Date(time))
    }

    fun timeFormatJustMMHH(time: Long): String {
        val pattern = StringBuffer()
        pattern.append("HH:mm")
        return SimpleDateFormat(pattern.toString()).format(Date(time))
    }
}
